File: Backend/StyleAura/src/main/java/com/styleaura/service/CategoryService.java
package com.styleaura.service;

import com.styleaura.entity.Category;

import java.util.List;

public interface CategoryService {

    List<Category> getAllCategories();

    Category getCategoryById(Long id);

    Category getCategoryBySlug(String slug);

    Category createCategory(Category category);

    Category updateCategory(Long id, Category category);

    void deleteCategory(Long id);
}

File: Frontend/src/services/categoryService.js
import axiosInstance from "./axiosInstance";

export const getAllCategories = () =>
    axiosInstance.get("/categories");

export const getCategoryById = (id) =>
    axiosInstance.get(`/categories/${id}`);

export const addCategory = (category) =>
    axiosInstance.post("/categories", category);

export const updateCategory = (id, category) =>
    axiosInstance.put(`/categories/${id}`, category);

export const deleteCategory = (id) =>
    axiosInstance.delete(`/categories/${id}`);